tests/hypervisor/test_local.py

```diff
@@ -5,40 +5,114 @@
+@pytest.mark.usefixtures("class_globalconf_clean")
+class TestLocalPositive:
+    @pytest.mark.tier1
+    def test_threads_number_after_reconfig_libvirt(self, virtwho, ssh_host):
+        """
+        :title: virt-who: loacl: test threads number after reconfig libvirt config file
+        :id: 4d2552ce-af14-4e4f-84ab-64764db64fe2
         :caseimportance: High
         :tags: tier1
         :customerscenario: false
         :upstream: no
         :steps:
-            1.
+            1. Run virt-who service to check the thread number
+            2. Update libvirt config, and restart libvirtd service
+            3. Check virt-who thread_num is changed or not
+            4. Recovery libvirt config
+
         :expectedresults:
-            1.
+            1. Succeeded to start virt-who service and the thread_num is 1
+            2. Succeed to start the libvirtd service
+            3. the thread_num is still 1
         """
-        logger.info("Succeeded to run the 'test_hostname_option'")
 
-    @pytest.mark.tier2
-    def test_http_option(self):
-        """Just a demo
+        # run virt-who service to check the thread number
+        result = virtwho.run_service()
+        thread_berfore = result["thread"]
+        logger.info(
+            f"Succeeded to start virt-who service and the thread_num is {thread_berfore}"
+        )
 
-        :title: virt-who: local: test http option
-        :id: a66787b3-1a5f-4b6f-9c09-0873d6490de3
-        :caseimportance: High
-        :tags: tier2
-        :customerscenario: false
-        :upstream: no
-        :steps:
-            1.
-        :expectedresults:
-            1.
-        """
-        logger.info("Succeeded to run the 'test_http_option'")
+        # update libvirt config, and restart libvirtd service
+        libvirt_conf = "/etc/libvirt/libvirtd.conf"
+        option_enable("listen_tls", libvirt_conf, ssh_host)
+        option_enable("listen_tcp", libvirt_conf, ssh_host)
+        option_enable("auth_tcp", libvirt_conf, ssh_host)
+        option_enable("tcp_port", libvirt_conf, ssh_host)
+
+        run_service(ssh_host, "libvirtd", "restart")
+        _, output = run_service(ssh_host, "libvirtd", "status")
+        assert "is running" in output or "Active: active (running)" in output
+
+        # check virt-who thread_num is changed or not
+        thread_after = virtwho.thread_number()
+        assert thread_berfore == thread_after
+
+        # recovery libvirt config
+        option_disable("listen_tls", libvirt_conf, ssh_host)
+        option_disable("listen_tcp", libvirt_conf, ssh_host)
+        option_disable("auth_tcp", libvirt_conf, ssh_host)
+        option_disable("tcp_port", libvirt_conf, ssh_host)
+
+        run_service(ssh_host, "libvirtd", "restart")
+        _, output = run_service(ssh_host, "libvirtd", "status")
+        assert "is running" in output or "Active: active (running)" in output
+
+
+def option_enable(option, file, ssh_host):
```

Comment:
do you think it's better to combine the two `enable` and `disable` functions together, it seems the params are same, and the code are similar.

Reply:
Good idea